bugsnag-android-performance/src/main/kotlin/com/bugsnag/android/performance/PerformanceConfiguration.kt

```diff
@@ -122,43 +133,60 @@ public class PerformanceConfiguration private constructor(public val context: Co
-                    ?.also { autoInstrumentAppStarts = it }
-                data?.getString(AUTO_INSTRUMENT_ACTIVITIES_KEY)
-                    ?.also { autoInstrumentActivities = AutoInstrument.valueOf(it) }
+                data.getString(AUTO_INSTRUMENT_ACTIVITIES_KEY)
+                    ?.also { config.autoInstrumentActivities = AutoInstrument.valueOf(it) }
 
                 // releaseStage / enabledReleaseStage
-                data?.getString(RELEASE_STAGE_KEY, data.getString(BSG_RELEASE_STAGE_KEY))
-                    ?.also { releaseStage = it }
-                data?.getString(ENABLED_RELEASE_STAGES, data.getString(BSG_ENABLED_RELEASE_STAGES))
-                    ?.also { enabledReleaseStages = it.splitToSequence(',').toSet() }
-
-                if (data?.containsKey(VERSION_CODE_KEY) == true) {
-                    versionCode = data.getInt(VERSION_CODE_KEY).toLong()
-                } else if (data?.containsKey(BSG_VERSION_CODE_KEY) == true) {
-                    versionCode = data.getInt(BSG_VERSION_CODE_KEY).toLong()
+                data.getString(RELEASE_STAGE_KEY, data.getString(BSG_RELEASE_STAGE_KEY))
+                    ?.also { config.releaseStage = it }
+                data.getString(ENABLED_RELEASE_STAGES, data.getString(BSG_ENABLED_RELEASE_STAGES))
+                    ?.also { config.enabledReleaseStages = it.splitToSequence(',').toSet() }
+
+                if (data.containsKey(VERSION_CODE_KEY)) {
+                    config.versionCode = data.getInt(VERSION_CODE_KEY).toLong()
+                } else if (data.containsKey(BSG_VERSION_CODE_KEY)) {
+                    config.versionCode = data.getInt(BSG_VERSION_CODE_KEY).toLong()
                 }
 
-                if (data?.containsKey(APP_VERSION_KEY) == true) {
-                    appVersion = data.getString(APP_VERSION_KEY)
-                } else if (data?.containsKey(BSG_APP_VERSION_KEY) == true) {
-                    appVersion = data.getString(BSG_APP_VERSION_KEY)
+                if (data.containsKey(APP_VERSION_KEY)) {
+                    config.appVersion = data.getString(APP_VERSION_KEY)
+                } else if (data.containsKey(BSG_APP_VERSION_KEY)) {
+                    config.appVersion = data.getString(BSG_APP_VERSION_KEY)
                 }
-            }
-        }
-    }
 
-    public fun addOnSpanEndCallback(spanEndCallback: SpanEndCallback) {
-        spanEndCallbacks.add(spanEndCallback)
-    }
+                if (data.containsKey(TRACE_PROPAGATION_URLS_KEY)) {
+                    config.tracePropagationUrls = data.getString(TRACE_PROPAGATION_URLS_KEY)
+                        ?.splitToSequence(',')
+                        ?.mapNotNull { pattern ->
+                            try {
+                                Pattern.compile(pattern)
+                            } catch (pse: PatternSyntaxException) {
+                                null
```

Comment:
Worth a `logger` message?

Reply:
🤔  on the notifier we actually let this through as an exception, and given that this is config maybe that is more appropriate?